packages/arb-token-bridge-ui/src/hooks/useOftTransactionHistory.ts

```diff
@@ -0,0 +1,352 @@
+import { useCallback } from 'react'
+import useSWR from 'swr'
+import { AssetType } from './arbTokenBridge.types'
+import { MergedTransaction } from '../state/app/state'
+import { getChainIdFromEid } from '../token-bridge-sdk/oftUtils'
+import { isDepositMode } from '../util/isDepositMode'
+import { useAccount } from 'wagmi'
+import { getProviderForChainId } from '../token-bridge-sdk/utils'
+import { fetchErc20Data } from '../util/TokenUtils'
+import { ethers, utils } from 'ethers'
+import { isNetwork } from '../util/networks'
+
+const LAYERZERO_API_URL_MAINNET = 'https://scan.layerzero-api.com/v1'
+const LAYERZERO_API_URL_TESTNET = 'https://scan-testnet.layerzero-api.com/v1'
+
+export enum LayerZeroMessageStatus {
+  INFLIGHT = 'INFLIGHT',
+  DELIVERED = 'DELIVERED',
+  FAILED = 'FAILED',
+  PAYLOAD_STORED = 'PAYLOAD_STORED',
+  BLOCKED = 'BLOCKED',
+  CONFIRMING = 'CONFIRMING'
+}
```

Comment:
looking at how it is used, i think using a `const` seems more suitable?

i.e.
```typescript
export const LayerZeroMessageStatus = {
  INFLIGHT: 'pending',
  CONFIRMING: 'pending',
  PAYLOAD_STORED: 'pending',
  DELIVERED: 'success',
  FAILED: 'failed',
  BLOCKED: 'failed'
} as const
```

then something like this:

```typescript
const getOftTransactionStatus = (message: LayerZeroMessage) => {
  return (
    LayerZeroMessageStatus[
      message.status.name as keyof typeof LayerZeroMessageStatus
    ] ?? 'pending'
  )
}
```

Reply:
Updated